test/TritonGPU/loop-pipeline.mlir

```diff
@@ -839,9 +839,16 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 4 :
     %14 = tt.broadcast %11 : tensor<1x16x!tt.ptr<f16>, #blocked> -> tensor<64x16x!tt.ptr<f16>, #blocked>
     %15 = tt.broadcast %13 : tensor<64x1xi32, #blocked> -> tensor<64x16xi32, #blocked>
     %16 = tt.addptr %14, %15 : tensor<64x16x!tt.ptr<f16>, #blocked>, tensor<64x16xi32, #blocked>
-    // check that the load didn't get pipelined.
-    // COMMON-NOT: alloc
-    // COMMON: scf.for
+    // check that the load with incompatiable shared encoding gets cloned and feeds into uses with same encoding
+    // AMD-NOT: alloc
+    // AMD: scf.for
+    // CHECK: local_alloc
+    // CHECK: local_alloc
+    // CHECK: scf.for
+    // CHECK: local_load {{.*}} tensor<64x16xf16, #triton_gpu.dot_op<{opIdx = 1
```

Comment:
This checks for single local_load only, why is that the case?

Reply:
The load has two uses, one is convert_layout, the other is local_alloc + trans.
Here local_load is for convert_layout usage, there is a later check for tt.trans that directly operates on memdesc, which is the second usage.
@pawelszczerbuk I will address the other comments soon.

Reply:
Could you share the IR after the pipelining? I thought we will clone the load, so we will end up with two local_loads

Reply:
Yes we will clone the load. See https://pastebin.com/GixMyv81
One of the use for the load is
```
       %24 = triton_gpu.local_alloc %18 : (tensor<64x16xf16, #blocked>) -> !tt.memdesc<64x16xf16, #shared, #triton_gpu.shared_memory>
      %25 = tt.trans %24 {order=array<i32: 1,0>} : !tt.memdesc<64x16xf16, #shared, #triton_gpu.shared_memory> -> !tt.memdesc<16x64xf16, #shared1, #triton_gpu.shared_memory>
```
This gets lowered to tt.trans without the local_load.

Reply:
Ah, I see the local_load is folded into local_alloc, makes sense. It would be great to verify in the test that the load really gets cloned, but I see how this can be hard with IR after the expander. I think I am good with the test!